src/core/manager.c

```diff
@@ -3828,8 +3828,10 @@ static int manager_run_generators(Manager *m) {
                 goto finish;
         }
 
+        /* Fork into a new user namespace, so that we don't get a permission error in case we are running inside
+         * an unprivileged container. */
         r = safe_fork("(sd-gens)",
-                      FORK_RESET_SIGNALS | FORK_WAIT | FORK_NEW_MOUNTNS | FORK_MOUNTNS_SLAVE | FORK_PRIVATE_TMP,
+                      FORK_RESET_SIGNALS | FORK_WAIT | FORK_NEW_MOUNTNS | FORK_MOUNTNS_SLAVE | FORK_PRIVATE_TMP | FORK_NEW_USERNS,
                       NULL);
```

Comment:
this is not going to suffice. this will not set up a uid mapping, you still need to do that in the child. did you test this?

Reply:
ah right, I thought we did that automatically when passing that flag